src/Core/Util/String/StringModifier.php

```diff
@@ -57,4 +64,40 @@ public function cutEnd(string $string, int $expectedLength): string
 
         return $string;
     }
+
+    /**
+     * Return a friendly url made from the provided string
+     * If the mbstring library is available, the output is the same as the js function of the same name.
+     *
+     * @param string $string
+     *
+     * @return string
+     */
+    public function str2url(string $string): string
+    {
+        $return_str = trim($string);
+        $return_str = mb_strtolower($return_str, 'UTF-8');
+
+        $return_str = $this->replaceAccentedChars($return_str);
+        $return_str = preg_replace('/[^a-zA-Z0-9\s\'\:\/\[\]\-\p{L}]/u', '', $return_str);
+        $return_str = preg_replace('/[\s\'\:\/\[\]\-]+/', ' ', $return_str);
+
+        return str_replace([' ', '/'], '-', $return_str);
+    }
+
+    /**
+     * Replace all accented chars by their equivalent non-accented chars.
+     *
+     * @param string $string
+     *
+     * @return string
+     */
+    public function replaceAccentedChars(string $string): string
+    {
+        if (null === $this->transliterator) {
+            $this->transliterator = Transliterator::create('Any-Latin; Latin-ASCII');
```

Comment:
I've already approved, but Im not sure if there is a point having transliterator here as a class property? I think it could be replaced by local variable instead.
Different transliterators might need to be used in different methods anyway. So Im not if there is a point having one as a class property.

Reply:
I kept the same behaviour as the previous code, and it also avoids to create an object every time you call this function.

To be honest I don't like this here, but it is better than the legacy code we have, and it's tested at least, so refactoring won't be difficult.